userland/test/CMakeLists.txt

```diff
@@ -4,12 +4,14 @@ foreach(middleware_impl_tmp ${middleware_implementations})
 
   add_executable(test_publisher__${middleware_impl_tmp} "test_publisher.cpp")
   ament_target_dependencies(test_publisher__${middleware_impl_tmp}
+    "rcl_interfaces"
```

Comment:
Why does this need to declare a dependency on `rcl_interfaces`? It is not using it directly. If it is a transitive requirement for `rclcpp` then `rclcpp` should export a dependency on `rcl_interfaces`.